fix issue: tag not creating in time

piggona opened the pull request.

piggona commented:
✗ make testacc TEST=./tencentcloud TESTARGS=" -run=TestAccTencentCloudAsScalingGroup_full"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./tencentcloud -v -run=TestAccTencentCloudAsScalingGroup_full -timeout 120m
=== RUN   TestAccTencentCloudAsScalingGroup_full
--- PASS: TestAccTencentCloudAsScalingGroup_full (70.95s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     71.025s

piggona commented:
make testacc TEST=./tencentcloud TESTARGS=" -run=TestAccTencentCloudCbsStorage_full" 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./tencentcloud -v -run=TestAccTencentCloudCbsStorage_full -timeout 120m
=== RUN   TestAccTencentCloudCbsStorage_full
=== PAUSE TestAccTencentCloudCbsStorage_full
=== CONT  TestAccTencentCloudCbsStorage_full
--- PASS: TestAccTencentCloudCbsStorage_full (6.02s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     6.085s

piggona commented:
✗ make testacc TEST=./tencentcloud TESTARGS=" -run=TestAccTencentCloudCfsFileSystem"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./tencentcloud -v -run=TestAccTencentCloudCfsFileSystem -timeout 120m
=== RUN   TestAccTencentCloudCfsFileSystemsDataSource
--- PASS: TestAccTencentCloudCfsFileSystemsDataSource (58.23s)
=== RUN   TestAccTencentCloudCfsFileSystem
--- PASS: TestAccTencentCloudCfsFileSystem (38.91s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     97.221s

piggona commented:
make testacc TEST=./tencentcloud TESTARGS=" -run=TestAccTencentCloudClbInstance_open"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./tencentcloud -v -run=TestAccTencentCloudClbInstance_open -timeout 120m
=== RUN   TestAccTencentCloudClbInstance_open
=== PAUSE TestAccTencentCloudClbInstance_open
=== CONT  TestAccTencentCloudClbInstance_open
--- PASS: TestAccTencentCloudClbInstance_open (24.25s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     24.321s

piggona commented:
make testacc TEST=./tencentcloud TESTARGS=" -run=TestAccTencentCloudCcnV3Basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./tencentcloud -v -run=TestAccTencentCloudCcnV3Basic -timeout 120m
=== RUN   TestAccTencentCloudCcnV3Basic
--- PASS: TestAccTencentCloudCcnV3Basic (17.02s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     17.085s

piggona commented:
✗ make testacc TEST=./tencentcloud TESTARGS=" -run=TestAccTencentCloudRedisInstance"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./tencentcloud -v -run=TestAccTencentCloudRedisInstance -timeout 120m
=== RUN   TestAccTencentCloudRedisInstancesDataSource
--- PASS: TestAccTencentCloudRedisInstancesDataSource (57.78s)
=== RUN   TestAccTencentCloudRedisInstance
--- PASS: TestAccTencentCloudRedisInstance (165.23s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     223.084s

Sherlock-Holo commented on tencentcloud/resource_tc_as_scaling_group.go:
```diff
@@ -431,11 +431,17 @@ func resourceTencentCloudAsScalingGroupRead(d *schema.ResourceData, meta interfa
+		return err
+	}
+
+	if err := d.Set("tags", tags); err != nil {
+		log.Printf("[CRITAL]%s provider set tags fail, reason:%s\n ", logId, err.Error())
+		return err
```
这个可以 _ = d.Set(tags) 忽略掉 error，一般不会出错

piggona replied:
fixed

Sherlock-Holo commented on tencentcloud/resource_tc_cbs_storage.go:
```diff
@@ -289,6 +284,18 @@ func resourceTencentCloudCbsStorageRead(d *schema.ResourceData, meta interface{}
+		return err
+	}
+
+	if err := d.Set("tags", tags); err != nil {
+		log.Printf("[CRITAL]%s provider set tags fail, reason:%s\n ", logId, err.Error())
+		return err
+	}
```
同上

Sherlock-Holo commented on tencentcloud/resource_tc_ccn.go:
```diff
@@ -137,6 +153,17 @@ func resourceTencentCloudCcnRead(d *schema.ResourceData, meta interface{}) error
+		return err
+	}
+
+	if err := d.Set("tags", tags); err != nil {
+		log.Printf("[CRITAL]%s provider set tags fail, reason:%s\n ", logId, err.Error())
+		return err
```
同上

Sherlock-Holo commented on tencentcloud/resource_tc_cfs_file_system.go:
```diff
@@ -220,6 +233,18 @@ func resourceTencentCloudCfsFileSystemRead(d *schema.ResourceData, meta interfac
+		return err
+	}
+
+	if err := d.Set("tags", tags); err != nil {
+		log.Printf("[CRITAL]%s provider set tags fail, reason:%s\n ", logId, err.Error())
+		return err
```
同上

Sherlock-Holo commented on tencentcloud/resource_tc_clb_instance.go:
```diff
@@ -336,7 +335,18 @@ func resourceTencentCloudClbInstanceRead(d *schema.ResourceData, meta interface{
+		return err
+	}
+
+	if err := d.Set("tags", tags); err != nil {
+		log.Printf("[CRITAL]%s provider set tags fail, reason:%s\n ", logId, err.Error())
+		return err
```
同上

piggona commented:
1. added: schema ‘tag' as TypeMap, involving resources: CCN, CFS
2. fixed: tag not creating/updating in time, involving resources: AS, CBS, CLB, Redis

oliverpei commented on tencentcloud/resource_tc_ccn.go:
```diff
@@ -173,6 +196,21 @@ func resourceTencentCloudCcnUpdate(d *schema.ResourceData, meta interface{}) err
+
+		tcClient := meta.(*TencentCloudClient).apiV3Conn
+		tagService := &TagService{client: tcClient}
+		region := meta.(*TencentCloudClient).apiV3Conn.Region
```
region := tcClient.Region

piggona replied:
fixed

oliverpei commented on tencentcloud/resource_tc_ccn.go:
```diff
@@ -173,6 +196,21 @@ func resourceTencentCloudCcnUpdate(d *schema.ResourceData, meta interface{}) err
+		if err != nil {
+			return err
+		}
+		d.SetPartial("tags")
```
这个需要先 d.Partial(true)

piggona replied:
fixed

oliverpei commented on tencentcloud/resource_tc_cfs_file_system_test.go:
```diff
@@ -119,5 +172,10 @@ resource "tencentcloud_cfs_file_system" "foo" {
   protocol = "NFS"
   vpc_id = tencentcloud_vpc.vpc.id
   subnet_id = tencentcloud_subnet.subnet.id
+  tags = {
```
了解下TF规范，块之间加个空行

piggona replied:
fixed
